### PR TITLE
pallet macro broke `benchmarks_instance`, fix by introducing  `benchmarks_instance_pallet`

### PR DESCRIPTION
**Pull request by thiolliere**
Currently instantiable pallet define with pallet macro cannot use benchmarks_instance because it bounds `I: Instance`.
Instead this PR introduce a new macro: `benchmarks_instance_pallet`

Also this PR make use of this macro for pallet-balances. Currently the benchmark only works for the default instance of pallet balance.

### CODE REVIEW

File: frame/benchmarking/src/lib.rs [hunk 1]
```diff
@@ -175,13 +175,33 @@ macro_rules! benchmarks {
...
+			( )
+			( )
+			$( $rest )*
+		);
+	}
+}
+
+/// Same as [`benchmarks`] but for instantiable pallet declared [`frame_support::pallet`].
+///
+/// NOTE: For pallet declared with `decl_module!`, use [`benchmarks_instnace`].
```
Comment (coriolinus) [inline frame/benchmarking/src/lib.rs:197]:
```suggestion
/// NOTE: For pallet declared with `decl_module!`, use [`benchmarks_instance`].
```

Typo

**Review by coriolinus — APPROVED** [review]

**Review by kianenigma — APPROVED** [review]
LGTM

**Comment by thiolliere** [comment]
@gnunicorn could we release frame-benchmarking 3.2 with this PR ?

**Comment by gnunicorn** [comment]
@thiolliere [the merge commit](https://github.com/paritytech/substrate/commit/6fae4a12d6205d4ae803df7e9afb4248b4ea3279) has already been released as 3.1:

image

**Comment by thiolliere** [comment]
damn I forgot, thank you.

**Pull request merged**